editor/plugins/packed_scene_translation_parser_plugin.cpp

```diff
@@ -52,32 +52,53 @@ Error PackedSceneEditorTranslationParserPlugin::parse_file(const String &p_path,
 	Ref<SceneState> state = Ref<PackedScene>(loaded_res)->get_state();
 
 	Vector<String> parsed_strings;
+	List<Pair<NodePath, bool>> atr_owners;
 	List<String> tabcontainer_paths;
 	for (int i = 0; i < state->get_node_count(); i++) {
 		String node_type = state->get_node_type(i);
-		bool is_control = ClassDB::is_parent_class(node_type, "Control");
-		if (!is_control && !ClassDB::is_parent_class(node_type, "Window")) {
-			continue;
-		}
+		String parent_path = state->get_node_path(i, true);
 
-		// Find the `auto_translate` property, and abort the string parsing of the node if disabled.
+		// Find the `auto_translate_mode` property, and abort the string parsing of the node if disabled, or inheriting a disabled one.
 		bool auto_translating = true;
 		for (int j = 0; j < state->get_node_property_count(i); j++) {
-			if (state->get_node_property_name(i, j) == "auto_translate" && (bool)state->get_node_property_value(i, j) == false) {
-				auto_translating = false;
-				break;
+			if (state->get_node_property_name(i, j) != "auto_translate_mode") {
+				continue;
 			}
+
+			int idx_last = atr_owners.size() - 1;
+			if (idx_last > 0 && !parent_path.begins_with(atr_owners[idx_last].first)) {
+				// Switch to the previous auto translation owner this was nested in, if that was the case.
+				atr_owners.pop_back();
+				idx_last -= 1;
+			}
+
+			int auto_translate_mode = (int)state->get_node_property_value(i, j);
+			if (auto_translate_mode == Control::AUTO_TRANSLATE_MODE_INHERIT) {
+				if (idx_last > 0 && atr_owners[idx_last].first == parent_path) {
+					auto_translating = atr_owners[idx_last].second;
+				} else {
+					atr_owners.push_back(Pair(state->get_node_path(i), true));
+				}
+			} else {
+				if (auto_translate_mode == Control::AUTO_TRANSLATE_MODE_DISABLED) {
+					auto_translating = false;
+				}
+
+				atr_owners.push_back(Pair(state->get_node_path(i), auto_translating));
+			}
+
+			break;
 		}
 
 		// Parse the names of children of `TabContainer`s, as they are used for tab titles.
 		if (!tabcontainer_paths.is_empty()) {
-			String parent_path = state->get_node_path(i, true);
 			if (!parent_path.begins_with(tabcontainer_paths[tabcontainer_paths.size() - 1])) {
 				// Switch to the previous `TabContainer` this was nested in, if that was the case.
 				tabcontainer_paths.pop_back();
 			}
 
-			if (is_control && auto_translating && !tabcontainer_paths.is_empty() && parent_path == tabcontainer_paths[tabcontainer_paths.size() - 1]) {
+			if (auto_translating && ClassDB::is_parent_class(node_type, "Control") && !tabcontainer_paths.is_empty() &&
```

Comment:
```suggestion
			if (auto_translating && !tabcontainer_paths.is_empty() && ClassDB::is_parent_class(node_type, "Control") &&
```
Let's put a cheaper condition first